Incompatible Typescript `Cache` type when using `createInMemoryCache`

### Describe the bug

When using the latest `@graphql-yoga/plugin-response-cache` plugin, providing a manually initialized `createInMemoryCache` cache instance causes a Typescript error:

```
error TS2322: Type 'import("/workspace/node_modules/@envelop/response-cache/typings/cache").Cache' is not assignable to type 'Cache'.
  The types returned by 'get(...)' are incompatible between these types.
    Type 'PromiseOrValue<ExecutionResult<ObjMap<unknown>, ObjMap<unknown>>>' is not assignable to type 'Promise<ExecutionResult<Record<string, unknown>, ResponseCachePluginExtensions>>'.
      Type 'ExecutionResult<ObjMap<unknown>, ObjMap<unknown>>' is missing the following properties from type 'Promise<ExecutionResult<Record<string, unknown>, ResponseCachePluginExtensions>>': then, catch, finally, [Symbol.toStringTag]
```

Prior versions of the plugin (1.4.0) did not cause this error.

### Your Example Website or App

https://codesandbox.io/p/sandbox/laughing-keldysh-ffhp3k?file=%2Findex.ts%3A4%2C20

### Steps to Reproduce the Bug or Issue

Create a Typescript instance of Yoga using the `useResponseCache` plugin, with the `cache` option set to a cache instance created with `createInMemoryCache`.

### Expected behavior

The server should start normally.

### Screenshots or Videos

_No response_

### Platform

Typescript

### Additional context

_No response_

--- a/.changeset/hungry-numbers-burn.md
+++ b/.changeset/hungry-numbers-burn.md
@@ -1,0 +1,5 @@
+---
+'@graphql-yoga/plugin-response-cache': patch
+---
+
+Fix incompatible type of in-memory cache factory with `cache` option

--- a/packages/plugins/response-cache/__tests__/response-cache.spec.ts
+++ b/packages/plugins/response-cache/__tests__/response-cache.spec.ts
@@ -1,6 +1,6 @@
 import { createSchema, createYoga } from 'graphql-yoga';
 import { cacheControlDirective } from '@envelop/response-cache';
-import { useResponseCache } from '@graphql-yoga/plugin-response-cache';
+import { createInMemoryCache, useResponseCache } from '@graphql-yoga/plugin-response-cache';
 
 const schema = createSchema({
   typeDefs: /* GraphQL */ `
@@ -390,4 +390,62 @@ it('should work with @cacheControl directive', async () => {
       },
     },
   });
+});
+
+it('should allow to create the cache outside of the plugin', async () => {
+  const onEnveloped = jest.fn();
+  const yoga = createYoga({
+    schema,
+    plugins: [
+      useResponseCache({
+        session: () => null,
+        includeExtensionMetadata: true,
+        cache: createInMemoryCache(),
+      }),
+      {
+        onEnveloped,
+      },
+    ],
+  });
+  const response = await yoga.fetch('http://localhost:3000/graphql', {
+    method: 'POST',
+    headers: {
+      'content-type': 'application/json',
+    },
+    body: JSON.stringify({ query: '{ _ }' }),
+  });
+
+  expect(response.status).toEqual(200);
+  const body = await response.json();
+  expect(body).toEqual({
+    data: {
+      _: 'DUMMY',
+    },
+    extensions: {
+      responseCache: {
+        didCache: true,
+        hit: false,
+        ttl: null,
+      },
+    },
+  });
+  const response2 = await yoga.fetch('http://localhost:3000/graphql', {
+    method: 'POST',
+    headers: {
+      'content-type': 'application/json',
+    },
+    body: JSON.stringify({ query: '{ _ }' }),
+  });
+  const body2 = await response2.json();
+  expect(body2).toMatchObject({
+    data: {
+      _: 'DUMMY',
+    },
+    extensions: {
+      responseCache: {
+        hit: true,
+      },
+    },
+  });
+  expect(onEnveloped).toHaveBeenCalledTimes(1);
 });

--- a/packages/plugins/response-cache/src/index.ts
+++ b/packages/plugins/response-cache/src/index.ts
@@ -2,11 +2,12 @@ import { ExecutionResult } from 'graphql';
 import { Maybe, Plugin, PromiseOrValue, YogaInitialContext, YogaLogger } from 'graphql-yoga';
 import {
   BuildResponseCacheKeyFunction,
-  createInMemoryCache,
   defaultBuildResponseCacheKey,
   Cache as EnvelopCache,
+  createInMemoryCache as envelopCreateInMemoryCache,
   ResponseCacheExtensions as EnvelopResponseCacheExtensions,
   GetDocumentStringFunction,
+  InMemoryCacheParameter,
   useResponseCache as useEnvelopResponseCache,
   UseResponseCacheParameter as UseEnvelopResponseCacheParameter,
 } from '@envelop/response-cache';
@@ -160,4 +161,7 @@ export function useResponseCache(options: UseResponseCacheParameter): Plugin {
   };
 }
 
-export { createInMemoryCache };
+export { InMemoryCacheParameter } from '@envelop/response-cache';
+export function createInMemoryCache(params?: InMemoryCacheParameter): Cache {
+  return envelopCreateInMemoryCache(params) as Cache;
+}

--- a/packages/plugins/response-cache/src/index.ts
+++ b/packages/plugins/response-cache/src/index.ts
@@ -7,7 +7,6 @@ import {
   createInMemoryCache as envelopCreateInMemoryCache,
   ResponseCacheExtensions as EnvelopResponseCacheExtensions,
   GetDocumentStringFunction,
-  InMemoryCacheParameter,
   useResponseCache as useEnvelopResponseCache,
   UseResponseCacheParameter as UseEnvelopResponseCacheParameter,
 } from '@envelop/response-cache';
@@ -161,7 +160,6 @@ export function useResponseCache(options: UseResponseCacheParameter): Plugin {
   };
 }
 
-export { InMemoryCacheParameter } from '@envelop/response-cache';
-export function createInMemoryCache(params?: InMemoryCacheParameter): Cache {
-  return envelopCreateInMemoryCache(params) as Cache;
-}
+export const createInMemoryCache = envelopCreateInMemoryCache as (
+  ...args: Parameters<typeof envelopCreateInMemoryCache>
+) => Cache;

